packages/unigraph-dev-explorer/src/components/ExplorerHome.tsx

```diff
@@ -2,10 +2,8 @@ import React from "react";
 
 export default function ExplorerHome(){
     const [messages, setMessages]: [string[], Function] = React.useState(window.unigraph.backendMessages);
-    window.unigraph.backendConnection.onmessage = (msg) => {
-        window.unigraph.backendMessages = [msg.data, ...messages]
-        setMessages(window.unigraph.backendMessages)
-    }
+
+    window.unigraph.addMessageListener((array: string[]) => setMessages(array))
```

Comment:
heads up, in my branch I think I made a change here – I didn't remove the listener, but put it into react state because a 2nd render of this component was causing a duplicate message to be displayed in the UI

so there will be a minor merge conflict here

Reply:
I've rebased it on your branch (just added request component and discarded my old changes)